### ISSUE TITLE
React Aria Tooltips

### ISSUE THREAD

**Issue by mgenitempo** [adobe/react-spectrum#8658]
### Provide a general summary of the feature here

The current iteration of tooltip cannot be dismissed using a pointer alone. In order to meet WCAG [Success Criteria 1.4.13](https://w3.org/TR/WCAG21#content-on-hover-or-focus).

Allow a mouse or pointer user to hide the tooltip, ideally through a close button (unless the tooltip will never overlap other content). The tooltip should not re-appear on subsequent hovers -- a zoom user will have dificulty trying to center their view on a specific area without triggering the tooltip and obscuring content.

Location: packages/react-aria-components/src/Tooltip.tsx

### 🤔 Expected Behavior?

There is a close button on the tooltip that is only accessible with a pointer. Once dismissed with this close button the tooltip does not reappear on subsequent hover actions.

### 😯 Current Behavior

Currently a user must move the pointer off the tooltip or press ESC to dismiss the tooltip. The tooltip continues to reappear after being dismissed.

### 💁 Possible Solution

Add a pointer only [x] close button. [Possible solution](https://sarahmhigley.com/writing/tooltips-in-wcag-21/#:~:text=WCAG%202.1%3A%20dismissable%2C%20hoverable%2C%20and%20persistent).

### 🔦 Context

I would like to be able to dismiss the tooltip usng only my mouse/pointer without moving the pointer too far out of the viewable area when I am zoomed in to read content.

### 💻 Examples

https://sarahmhigley.com/writing/tooltips-in-wcag-21/#:~:text=WCAG%202.1%3A%20dismissable%2C%20hoverable%2C%20and%20persistent

### 🧢 Your Company/Team

React Aria

**Comment by snowystinger** [comment]
I suspect this is a bit of a grey area. @majornista please correct me if I misstate anything.

Tooltips cannot contain buttons, https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/tooltip_role
> Because the tooltip itself never receives focus and is not in the tabbing order, a tooltip can not contain interactive elements like links, inputs, or buttons.

I do see in the link to the aria success criteria though
> A [mechanism](https://www.w3.org/TR/WCAG21/#dfn-mechanism) is available to dismiss the additional content without moving pointer hover

However, there is slight difference in the phrasing there and what I think you're expecting, which says "without moving pointer hover". I think this can be interpreted as, the "Esc" key also works for a pointer user. The user doesn't need to move the pointer in order to dismiss the tooltip.

**Comment by joshkimux** [comment]
@snowystinger , hi friend! I was curious if y'all had done discovery on this point in [Sarah's breakdown of tooltips](https://sarahmhigley.com/writing/tooltips-in-wcag-21/):

> (*) An even more nitpicky side note: WCAG 1.4.13 says that a tooltip should be dismissable "without moving pointer hover or keyboard focus," then goes on to suggest the Escape key to fulfill this requirement. This makes sense for keyboard users, but not at all for mouse users. On a usability level, people who primarily rely on a mouse or pointer are less likely to know keyboard shortcuts, and may not be able to use them. On a technical level, it is impossible to capture an escape press on a control that is hovered but not focused. A global escape listener would not be able to differentiate between a user wanting to dismiss a tooltip vs. close a dialog, if one were open. In short, the only feasible solution is to provide both keyboard and pointer methods of dismissal.

We're particularly concerned about the usability issues she's highlighting here, especially given her prominence in the accessibility community.

**Comment by snowystinger** [comment]
I do not know that we have explicitly compared to this post. @majornista if you have more insight

> A global escape listener would not be able to differentiate between a user wanting to dismiss a tooltip vs. close a dialog, if one were open. In short, the only feasible solution is to provide both keyboard and pointer methods of dismissal.

I'm not sure this is entirely true. We do use a global listener for the tooltip, and we keep a stack of our overlays so we can know which one should close. In addition, because focus cannot leave the dialog (except to go to a new nested overlay or some other top level area like Toasts), we don't need to use a global listener for the Dialog, we can just listen to keyboard events that come through the Dialog itself. So there isn't a conflict there.
In addition, our global listener will stop the propagation of the keyboard even in the capture phase, so nothing else should ever handle the event or even be aware of it.

I do not know enough about usability for primary mouse/pointer users not knowing Escape is a keyboard "shortcut". Hopefully our accessibility team can chime in more on that point. @jnurthen

Technically there is another option available for users to forcibly dismiss the tooltip if they are a mouse/pointer user, they can click on any other part of the UI which will result in focus leaving.

Regardless of anything though, we cannot add a close button. That's just invalid ARIA at this point in time.

**Comment by joshkimux** [comment]
> I do not know enough about usability for primary mouse/pointer users not knowing Escape is a keyboard "shortcut". Hopefully our accessibility team can chime in more on that point. @jnurthen

Thank you for tagging in your accessibility team! We'd love to hear from them on this.
